src/main/java/org/folio/service/titles/TitlesService.java

```diff
@@ -48,12 +48,11 @@ public TitlesService(RestClient restClient, ProtectionService protectionService,
   }
 
   public Future<Title> createTitle(Title title, RequestContext requestContext) {
-    String targetTenantId = TenantTool.tenantId(requestContext.getHeaders());
     return protectionService.validateAcqUnitsOnCreate(title.getAcqUnitIds(), TITLES_ASSIGN, requestContext)
-      .compose(v -> inventoryInstanceManager.createShadowInstanceIfNeeded(title.getInstanceId(), targetTenantId, requestContext))
-      .compose(shadowInstance -> {
+      .compose(v -> inventoryInstanceManager.getOrCreateInstanceRecord(title, requestContext))
+      .compose(instId -> {
```

Comment:
I think we need the following logic here:

- firstly invoke createShadowInstanceIfNeeded(). In this case when we are in ECS and shared instance is exists in central tenant - we would not need to create instance with another id. Instead we will use shadow instance with the same instanceId as in central tenant 
- after this invoke inventoryInstanceManager.getOrCreateInstanceRecord() only in case when shadow instance from previous step was not created.
So for ECS envs we will use shadow instance, for non ECS - create new instance that is our desired result.
We can encapsulate this logic in TitleInstanceService

Reply:
Modified the logic